test/DynamoCoreTests/ProfilingTest.cs

```diff
@@ -94,5 +96,58 @@ public void TestProfilingSingleNodePublicMethodsOnly()
             Assert.IsNotNull(profilingData.NodeExecutionTime(node));
             Assert.Greater(profilingData.NodeExecutionTime(node)?.Ticks, 0);
         }
+
+        private static List<Guid> executingNodes = new List<Guid>();
+        private static int nodeExecutionBeginCount = 0;
+        private static int nodeExecutionEndCount = 0;
+
+        private static void onNodeExectuionBegin(NodeModel sender, NodeModel.NodeExecutionEventArgs args)
+        {
+            // Assert that the node has ot been executed more than once
+            CollectionAssert.DoesNotContain(executingNodes, args.GUID);
+
+            executingNodes.Add(args.GUID);
+            nodeExecutionBeginCount++;
+        }
+
+        private static void onNodeExectuionEnd(NodeModel sender, NodeModel.NodeExecutionEventArgs args)
+        {
+            // Assert that the node ending execution had the begin exectuion event fired previously
+            CollectionAssert.Contains(executingNodes, args.GUID);
+
+            nodeExecutionEndCount++;
+        }
+
+       [Test]
+        public void TestNodeExecutionEvents()
```

Comment:
can we add test that asserts the value of the `data` object? Or get rid of that property if we think it should not be passed.

Reply:
After looking into the tests for the data object it seemed to me that the data object was redundant because the NodeModel object was the important information which was already being passed in. I removed the class that contained the NodeModel GUID and the now extraneous data and just rely on the NodeModel object that was being passed through already.